debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/DebeziumContainer.java

```diff
@@ -84,45 +125,137 @@ private void registerConnectorToDebezium(final String payload, final String full
+                responseBody.close();
+            }
+            throw new IOException("Unexpected response: " + response + " Response Body: " + responseBodyContent);
         }
+        return response;
     }
 
-    /**
-     * Returns the "/connectors" endpoint.
-     */
-    public String getConnectors() {
-        return getTarget() + "/connectors/";
+    public boolean connectorIsNotRegistered(String connectorName) throws IOException {
+        try (Response response = executeRequest(new Request.Builder().url(getConnectorURI(connectorName)).get().build())) {
+            boolean connectorIsNotRegistered = response.code() == 404;
+            response.close();
+            return connectorIsNotRegistered;
+        }
     }
 
-    /**
-     * Returns the "/connectors/<connector>" endpoint.
-     */
-    public String getConnector(String connectorName) {
-        return getConnectors() + connectorName;
+    protected void deleteDebeziumConnector(String connectorName) throws IOException {
+        executeRequestSuccessful(new Request.Builder().url(getConnectorURI(connectorName)).delete().build()).close();
     }
 
-    /**
-     * Returns the "/connectors/<connector>/status" endpoint.
-     */
-    public String getConnectorStatus(String connectorName) {
-        return getConnectors() + connectorName + "/status";
+    public void deleteConnector(String connectorName) throws IOException {
+        deleteDebeziumConnector(connectorName);
+
+        Awaitility.await()
+                .atMost(10, TimeUnit.SECONDS)
+                .until(() -> connectorIsNotRegistered(connectorName));
     }
 
-    public String getTarget() {
-        return "http://" + getContainerIpAddress() + ":" + getMappedPort(8083);
+    public List<String> getRegisteredConnectors() throws IOException {
+        Request request = new Request.Builder().url(getConnectorsURI()).get().build();
+        try (ResponseBody responseBody = executeRequestSuccessful(request).body()) {
+            if (null != responseBody) {
+                String string = responseBody.string();
+                responseBody.close();
+                return MAPPER.readValue(string, new TypeReference<List<String>>() {
+                });
+            }
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return Collections.emptyList();
+    }
+
+    public boolean isConnectorConfigured(String connectorName) throws IOException {
+        Request request = new Request.Builder().url(getConnectorURI(connectorName)).get().build();
+        try (Response response = executeRequest(request)) {
+            boolean isConnectorConfigured = response.isSuccessful();
+            response.close();
+            return isConnectorConfigured;
+        }
+    }
+
+    public void ensureConnectorRegistered(String connectorName) {
+        Awaitility.await()
+                .atMost(10, TimeUnit.SECONDS)
```

Comment:
Configurable timeout

Reply:
The "old" code already has static timeouts and the 10 sec timeout is high enough to be on the safe for every serious CI env. If we see problems with this later, I think we can fix that later once we see issues with that.

Reply:
Done